Documents/Automation/EncryptRootVolume/Documents/npark-encryptrootvolume.json

```diff
@@ -184,15 +200,49 @@
       "inputs": {
         "DocumentName": "AWS-AttachEBSVolume",
         "RuntimeParameters": {
-          "Device": "{{devicename}}",
-          "InstanceId": "{{instanceId}}",
+          "Device": "{{describeInstance.RootDeviceName}}",
+          "InstanceId": "{{InstanceId}}",
           "VolumeId": "{{CreateVolume.NewRootVolumeID}}"
         }
       },
       "name": "AttachNewEBSVolume",
       "action": "aws:executeAutomation",
       "timeoutSeconds": 180,
       "onFailure": "Abort"
+    },
+    {
+       "name": "ApplyDeleteOnTerminationValue",
+       "action": "aws:executeAwsApi",
+       "onFailure": "Abort",
+       "timeoutSeconds": 60,
+       "maxAttempts": 10,
+       "inputs": {
+         "Service": "ec2",
+         "Api": "ModifyInstanceAttribute",
+         "InstanceId": "{{InstanceId}}",
+         "BlockDeviceMappings": [
+           {
+             "DeviceName": "{{describeInstance.RootDeviceName}}",
+             "Ebs": {
+               "DeleteOnTermination": "{{describeInstanceRootVolume.RootDeviceDeleteOnTermination}}"
+             }
+           }
+         ]
+       },
+       "isCritical": true
+    },
+    {
+      "maxAttempts": 1,
+      "inputs": {
+        "DocumentName": "AWS-StartEC2Instance",
```

Comment:
You are making an arbitrary decision to start the instance. Instead, get the current state when you describe the input instance, then use aws:changeInstanceState and re-apply the initial state here, i.e.

```json
{
      "name": "restoreInstanceInitialState",
      "action": "aws:changeInstanceState",
      "onFailure": "Abort",
      "inputs": {
        "InstanceIds": [
          "{{ InstanceId }}"
        ],
        "DesiredState": "{{ describeInstance.State }}"
      },
      "isCritical": true
}
```

Reply:
Done.